### ISSUE TITLE
v1.7.29 Stopped recording prometheus Metrics data

### ISSUE THREAD

**Issue by jacobrandall** [traefik/traefik#8039]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Pulled the latest `traefik:1.7` docker image which now points at `v1.7.29`. (Previously it was pointing at `v1.7.28` and earlier)

### What did you expect to see?

Prometheus metrics for traefik pods

### What did you see instead?

No traefik prometheus metrics for traefik pods. To clarify, we can still scrape the prometheus endpoint, but it is missing all of the `^traefik*` metrics like:

```
traefik_backend_server_up
traefik_config_last_reload_failure
traefik_config_last_reload_success
traefik_config_reloads_failure_total
traefik_config_reloads_total
traefik_entrypoint_open_connections
traefik_entrypoint_request_duration_seconds_bucket
traefik_entrypoint_request_duration_seconds_count
traefik_entrypoint_request_duration_seconds_sum
traefik_entrypoint_requests_total
```

We only see metrics for the following:

```
go_gc_duration_seconds
go_gc_duration_seconds_count
go_gc_duration_seconds_sum
go_goroutines
go_info
go_memstats_alloc_bytes
go_memstats_alloc_bytes_total
go_memstats_buck_hash_sys_bytes
go_memstats_frees_total
go_memstats_gc_cpu_fraction
go_memstats_gc_sys_bytes
go_memstats_heap_alloc_bytes
go_memstats_heap_idle_bytes
go_memstats_heap_inuse_bytes
go_memstats_heap_objects
go_memstats_heap_released_bytes
go_memstats_heap_sys_bytes
go_memstats_last_gc_time_seconds
go_memstats_lookups_total
go_memstats_mallocs_total
go_memstats_mcache_inuse_bytes
go_memstats_mcache_sys_bytes
go_memstats_mspan_inuse_bytes
go_memstats_mspan_sys_bytes
go_memstats_next_gc_bytes
go_memstats_other_sys_bytes
go_memstats_stack_inuse_bytes
go_memstats_stack_sys_bytes
go_memstats_sys_bytes
go_threads
process_cpu_seconds_total
process_max_fds
process_open_fds
process_resident_memory_bytes
process_start_time_seconds
process_virtual_memory_bytes
process_virtual_memory_max_bytes
promhttp_metric_handler_requests_in_flight
promhttp_metric_handler_requests_total
```

### Output of `traefik version`: 1.7.29

```
$ docker run traefik:1.7 version
Version:      v1.7.29
Codename:     maroilles
Go version:   go1.16.2
Built:        2021-03-29_04:26:01PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Traefik still runs as expected with the exception of prometheus metrics no longer being recorded. Using the same config but with an earlier version of traefik (`v1.7.24` - `v1.7.28`) confirms prometheus metrics are recorded in earlier versions.

This issue was first reported here https://github.com/traefik/traefik/issues/8032 but creating a new one with the bug template info that was requested on that ticket (I don't have permissions to edit that one, lmk if you prefer something else).

**Comment by jbdoumenjou** [comment]
Closed by #8040

**Comment by jacobrandall** [comment]
Awesome, thanks for the quick response 🙌 . We'll keep an eye out for a release and try it out.
